pkg/podvolume/backupper_factory.go

```diff
@@ -41,45 +40,46 @@ type BackupperFactory interface {
 func NewBackupperFactory(
 	repoLocker *repository.RepoLocker,
 	repoEnsurer *repository.Ensurer,
-	veleroClient clientset.Interface,
 	pvcClient corev1client.PersistentVolumeClaimsGetter,
 	pvClient corev1client.PersistentVolumesGetter,
 	podClient corev1client.PodsGetter,
+	crClient ctrlclient.Client,
+	crWatchClient ctrlclient.WithWatch,
 	log logrus.FieldLogger,
 ) BackupperFactory {
 	return &backupperFactory{
-		repoLocker:   repoLocker,
-		repoEnsurer:  repoEnsurer,
-		veleroClient: veleroClient,
-		pvcClient:    pvcClient,
-		pvClient:     pvClient,
-		podClient:    podClient,
-		log:          log,
+		repoLocker:    repoLocker,
+		repoEnsurer:   repoEnsurer,
+		pvcClient:     pvcClient,
+		pvClient:      pvClient,
+		podClient:     podClient,
+		crClient:      crClient,
+		crWatchClient: crWatchClient,
+		log:           log,
 	}
 }
 
 type backupperFactory struct {
-	repoLocker   *repository.RepoLocker
-	repoEnsurer  *repository.Ensurer
-	veleroClient clientset.Interface
-	pvcClient    corev1client.PersistentVolumeClaimsGetter
-	pvClient     corev1client.PersistentVolumesGetter
-	podClient    corev1client.PodsGetter
-	log          logrus.FieldLogger
+	repoLocker    *repository.RepoLocker
+	repoEnsurer   *repository.Ensurer
+	pvcClient     corev1client.PersistentVolumeClaimsGetter
+	pvClient      corev1client.PersistentVolumesGetter
+	podClient     corev1client.PodsGetter
+	crClient      ctrlclient.Client
+	crWatchClient ctrlclient.WithWatch
+	log           logrus.FieldLogger
 }
 
 func (bf *backupperFactory) NewBackupper(ctx context.Context, backup *velerov1api.Backup, uploaderType string) (Backupper, error) {
-	informer := velerov1informers.NewFilteredPodVolumeBackupInformer(
-		bf.veleroClient,
-		backup.Namespace,
-		0,
-		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
-		func(opts *metav1.ListOptions) {
-			opts.LabelSelector = fmt.Sprintf("%s=%s", velerov1api.BackupUIDLabel, backup.UID)
-		},
-	)
+	lw := kube.InternalLW{
+		Client:     bf.crWatchClient,
+		Namespace:  backup.Namespace,
+		ObjectList: new(velerov1api.PodVolumeBackupList),
+	}
+
+	informer := cache.NewSharedInformer(&lw, &velerov1api.PodVolumeBackup{}, time.Second)
```

Comment:
The controller manager manages Informers internally, let's get the informer via the manager instead, there will be unique informers for different kinds of resources globally.
`mgr.GetCache().GetInformer()`

Reply:
Thanks. I will modify it.